genai-perf/genai_perf/config/generate/perf_analyzer_config.py

```diff
@@ -0,0 +1,146 @@
+# limitations under the License.
+
+from typing import Any, Dict, List
+
+from genai_perf.config.generate.search_parameter import SearchUsage
+from genai_perf.config.input.config_command import ConfigCommand, ConfigPerfAnalyzer
+from genai_perf.types import ModelName, ModelObjectiveParameters
+
+
+class PerfAnalyzerConfig:
+    """
+    Contains the all the methods necessary for handling calls
+    to PerfAnalzyer
+    """
+
+    def __init__(
+        self,
+        config: ConfigCommand,
+        model_objective_parameters: ModelObjectiveParameters,
+        model_name: ModelName,
+    ):
+        self._model_name = model_name
+        self._set_options_based_on_config(config)
+        self._set_options_based_on_objective(model_objective_parameters)
+
+    ###########################################################################
+    # Set Options Methods
+    ###########################################################################
+    def _set_options_based_on_config(self, config: ConfigCommand) -> None:
+        self._config: ConfigPerfAnalyzer = config.perf_analzyer
+
+    def _set_options_based_on_objective(
+        self, model_objective_parameters: ModelObjectiveParameters
+    ) -> None:
+        self._parameters: Dict[str, Any] = {}
+        for objective in model_objective_parameters.values():
+            for name, parameter in objective.items():
+                if parameter.usage == SearchUsage.RUNTIME:
+                    self._parameters[name] = parameter.get_value_based_on_category()
+
+    ###########################################################################
+    # CLI String Creation Methods
+    ###########################################################################
+    def create_cli_string(self) -> str:
+        """
+        Returns the PA command as a string
+        """
+        cli_args = self._create_required_args()
+        cli_args += self._create_parameter_args()
+
+        cli_string = " ".join(cli_args)
+        return cli_string
+
+    def _create_required_args(self) -> List[str]:
+        # These come from the config and are always used
+        required_args = [
+            self._config.path,
+            "--model-name",
+            self._model_name,
+            "--stability-percentage",
+            str(self._config.stability_threshold),
+        ]
+
+        return required_args
+
+    def _create_parameter_args(self) -> List[str]:
+        parameter_args = []
+        for name, value in self._parameters.items():
+            parameter_args.append(self._convert_objective_to_cli_option(name))
+            parameter_args.append(str(value))
+
+        return parameter_args
+
+    def _convert_objective_to_cli_option(self, objective_name: str) -> str:
+        obj_to_cli_dict = {
+            "runtime_batch_size": "--batch-size",
+            "concurrency": "--concurrency-range",
+            "request-rate": "--request-rate-range",
+        }
+
+        return obj_to_cli_dict[objective_name]
```

Comment:
You might want to use error handling here if it is possible that the objective name is ever not in the dictionary.

Reply:
Good suggestion!